src/function/table/read_csv.cpp

```diff
@@ -355,94 +100,59 @@ void ReadCSVTableFunction::ReadCSVAddNamedParameters(TableFunction &table_functi
-	auto &bind_data = bind_data_p->Cast<ReadCSVData>();
-	// determined through the scientific method as the average amount of rows in a CSV file
-	idx_t per_file_cardinality = 42;
-	if (bind_data.buffer_manager && bind_data.buffer_manager->file_handle) {
-		auto estimated_row_width = (bind_data.csv_types.size() * 5);
-		per_file_cardinality = bind_data.buffer_manager->file_handle->FileSize() / estimated_row_width;
-	}
-	return make_uniq<NodeStatistics>(bind_data.files.size() * per_file_cardinality);
-}
-
 static void CSVReaderSerialize(Serializer &serializer, const optional_ptr<FunctionData> bind_data_p,
                                const TableFunction &function) {
-	auto &bind_data = bind_data_p->Cast<ReadCSVData>();
-	serializer.WriteProperty(100, "extra_info", function.extra_info);
-	serializer.WriteProperty(101, "csv_data", &bind_data);
+	throw NotImplementedException("CSVReaderSerialize not implemented");
+	// auto &bind_data = bind_data_p->Cast<MultiFileBindData>();
+	// auto &csv_data = bind_data.bind_data->Cast<ReadCSVData>();
+	// serializer.WriteProperty(100, "extra_info", function.extra_info);
+	// // generate the serialized data
+	// SerializedReadCSVData serialized_data;
+	// serialized_data.return_types = serialized_data.csv_types = bind_data.types;
+	// serialized_data.return_names = serialized_data.csv_names = bind_data.names;
+	// serialized_data.files = bind_data.file_list->GetAllFiles();
+	// serialized_data.filename_col_idx = csv_data.filename_col_idx;
+	// serialized_data.options.options = csv_data.options;
+	// serialized_data.options.file_options = bind_data.file_options;
+	// serialized_data.reader_bind = bind_data.reader_bind;
+	// serializer.WriteProperty(101, "csv_data", serialized_data);
 }
 
 static unique_ptr<FunctionData> CSVReaderDeserialize(Deserializer &deserializer, TableFunction &function) {
-	unique_ptr<ReadCSVData> result;
-	deserializer.ReadProperty(100, "extra_info", function.extra_info);
-	deserializer.ReadProperty(101, "csv_data", result);
-	return std::move(result);
-}
-
-void PushdownTypeToCSVScanner(ClientContext &context, optional_ptr<FunctionData> bind_data,
-                              const unordered_map<idx_t, LogicalType> &new_column_types) {
-	auto &csv_bind = bind_data->Cast<ReadCSVData>();
-	for (auto &type : new_column_types) {
-		csv_bind.csv_types[type.first] = type.second;
-		csv_bind.return_types[type.first] = type.second;
-	}
-}
-
-virtual_column_map_t ReadCSVGetVirtualColumns(ClientContext &context, optional_ptr<FunctionData> bind_data) {
-	auto &csv_bind = bind_data->Cast<ReadCSVData>();
-	virtual_column_map_t result;
-	MultiFileReader::GetVirtualColumns(context, csv_bind.reader_bind, result);
-	result.insert(make_pair(COLUMN_IDENTIFIER_EMPTY, TableColumn("", LogicalType::BOOLEAN)));
-	return result;
+	throw NotImplementedException("CSVReaderDeserialize not implemented");
+	// auto &context = deserializer.Get<ClientContext &>();
```

Comment:
Should we remove the comments?

Reply:
I'll keep the comments here for now in case we need to implement the serialization - I think we don't need it anymore but let's see